tests/sentry/utils/test_committers.py

```diff
@@ -207,3 +215,77 @@ def test_simple(self):
         assert len(releases) == 2
         assert releases[0] == release2
         assert releases[1] == release1
+
+
+class GetEventFileCommitters(CommitTestCase):
+    def setUp(self):
+        super(GetEventFileCommitters, self).setUp()
+        release = self.create_release(
+            project=self.project,
+            version='v12'
+        )
+        release.set_commits([
+            {
+                'id': 'a' * 40,
+                'repository': self.repo.name,
+                'author_email': 'bob@example.com',
+                'author_name': 'Bob',
+                'message': 'i fixed a bug',
+                'patch_set': [
+                    {
+                        'path': 'sentry/example/Application/Application.java',
+                        'type': 'M',
+                    },
+                ]
+            }
+        ])
+        group = self.create_group(
+            project=self.project,
+            message='Kaboom!',
+            first_release=release,
+        )
+        self.event = self.create_event(
+            group=group,
+            message='Kaboom!',
+            platform='java',
+            data={
+                'sdk': {
+                    'name': 'sentry-java',
+                    'version': '1.0',
+                },
+            },
+            stacktrace={
+                'frames': [
+                    {
+                        "function": "invoke0",
+                        "abs_path": "NativeMethodAccessorImpl.java",
+                        "in_app": False,
+                        "module": "jdk.internal.reflect.NativeMethodAccessorImpl",
+                        "filename": "NativeMethodAccessorImpl.java",
+                    },
+                    {
+                        "function": "home",
+                        "abs_path": "Application.java",
+                        "module": "io.sentry.example.Application",
+                        "in_app": True,
+                        "lineno": 30,
+                        "filename": "Application.java",
+                    },
+                    {
+                        "function": "handledError",
+                        "abs_path": "Application.java",
+                        "module": "io.sentry.example.Application",
+                        "in_app": True,
+                        "lineno": 39,
+                        "filename": "Application.java",
+                    }
+                ]
+            }
+        )
+
+    def test_java_sdk_path_mangling(self):
+        result = get_event_file_committers(self.project, self.event)
```

Comment:
Could you add tests for your changes as well?

Reply:
This test is for my change. There are no tests for non-java platforms :cry:. I can add some more coverage for those paths though.

Reply:
I think I'm misunderstanding what this is testing... Does using these function calls inherently test that the paths are formatted as you expect?

Reply:
This test covers the new path mangling logic for java frames. Without that code this test will fail as the commit will not match the frame filepath. Our matching process requires at two or more common path segments to result in a hit, which was not possible before.